taskvine/src/bindings/python3/ndcctools/taskvine/futures.py

```diff
@@ -45,16 +49,34 @@ def __init__(self, port=9123, batch_type="local", manager=None, manager_host_por
         else: 
             self.factory = None
 
-    def submit(self, fn, *args, **kwargs):
-        if isinstance(fn, FutureTask):
-            self.manager.submit(fn)
-            return fn._future
-        future_task = FutureTask(self.manager, False, fn, *args, **kwargs)
-        self.manager.submit(future_task)
-        return future_task._future
+    def submit(self, task):
+        if isinstance(task, FuturePythonTask):
+            self.manager.submit(task)
+            return task._future
+        if isinstance(task, FutureFunctionCall):
+            # submit the task
+            self.manager.submit(task)
+            # append it to the task table
+            self.task_table.append(task)
+            return
```

Comment:
When we submit a `FutureFunctionCall` it should return a `VineFuture`. So, a `FutureFunctionCall` should have a future field.